pymeasure/instruments/velleman/velleman_k8090.py

```diff
@@ -0,0 +1,272 @@
+    ==================  ==================
+
+    Use the class like:
+
+    .. code-block:: python
+
+       from pymeasure.instruments.velleman import VellemanK8090, VellemanK8090Switches as Switches
+       from pymeasure.adapters import SerialAdapter
+
+       adapter = SerialAdapter("COM1", baudrate=19200, timeout=1.0)
+       instrument = VellemanK8090(adapter)
+
+       # Get status update from device
+       last_on, curr_on, time_on = instrument.status
+
+       # Toggle a selection of channels on
+       instrument.switch_on = Switches.CH3 | Switches.CH4 | Switches.CH5
+
+    """
+
+    def __init__(self, adapter, name="Velleman K8090", timeout=1000, **kwargs):
+        super().__init__(
+            adapter,
+            name=name,
+            asrl={"baud_rate": 19200},
+            timeout=timeout,
+            **kwargs
+        )
+
+    BYTE_STX = 0x04
+    BYTE_ETX = 0x0F
+
+    version = Instrument.measurement(
+        "0x71",
+        """
+        Get firmware version, as (year - 2000, week). E.g. ``(10, 1)``
+        """,
+        cast=int,
+        get_process=lambda v: (v[2], v[3]) if len(v) > 3 and v[0] == 0x71 else None,
+    )
+
+    status = Instrument.measurement(
+        "0x18",
+        """
+        Get current relay status.
+        The reply has a different command byte than the request.
+
+        Three items (:class:`VellemanK8090Switches` flags) are returned:
+
+        * Previous state: the state of each relay before this event
+        * Current state: the state of each relay now
+        * Timer state: the state of each relay timer
+        """,
+        cast=int,
+        get_process=_get_process_status,
+    )
+
+    switch_on = Instrument.setting(
+        "0x11,%s",
+        """"
+        Switch on a set of channels.
+        Other channels are unaffected.
+        Pass either a list or set of channel numbers (starting at 1), or pass a bitmask
+        """,
+        set_process=_parse_channels,
+    )
+
+    switch_off = Instrument.setting(
+        "0x12,%s",
+        """
+        Switch off a set of channels. See :attr:`switch_on`.
+        """,
+        set_process=_parse_channels,
+    )
+
+    id = None  # No identification available
+
+    def _make_checksum(self, command, mask, param1, param2):
+        # The formula from the sheet requires twos-complement negation,
+        # this works
+        return 1 + 0xFF - ((self.BYTE_STX + command + mask + param1 + param2) & 0xFF)
+
+    def switch_on_blocking(self, channels):
+        """Switch on a set of channels and wait for the confirmation.
+
+        See :attr:`switch_on`.
+
+        The set command is identical, and the processing part of
+        :attr:`status` is called.
+
+        :returns: See :attr:`status`
+        """
+        mask = _parse_channels(channels)
+        self.write(f"0x11,{mask}")
+        data = self.read()
+        items = [int(it) for it in data.split(",")]
+        return _get_process_status(items)
```

Comment:
In order to not repeat yourself, I'd access the already defined methods:
```suggestion
        self.switch_on = channels
        return self.status
```

Reply:
I should indeed use `self.switch_on` here, but we cannot directly use `self.status` as that also requests a status, causing a duplicate request-response couples.
The point of this method is to capture that automated response.

Reply:
ah, I see.

Reply:
I dropped these blocking versions completely, because the K8090 replies make this a huge mess.